src/multi_physics/QED/src/containers/picsar_tables.hpp

```diff
@@ -0,0 +1,718 @@
+            m_x_min{x_min}, m_x_max{x_max}, m_values{values}
+            {
+                m_how_many_x = static_cast<int>(values.size());
+                m_x_size = x_max - x_min;
+            };
+
+        /**
+        * Empty constructor
+        */
+        PXRMP_INTERNAL_GPU_DECORATOR PXRMP_INTERNAL_FORCE_INLINE_DECORATOR
+        equispaced_1d_table(){};
+
+        /**
+        * Constructor from byte array (not usable on GPUs)
+        *
+        * @param[in] raw_data a const reference to a byte vector
+        */
+        equispaced_1d_table(const std::vector<char>& raw_data)
+        {
+            using namespace picsar::multi_physics::utils;
+
+            constexpr size_t min_size =
+                sizeof(char)+//double or float
+                sizeof(m_x_min)+sizeof(m_x_max)+
+                sizeof(m_how_many_x);
+
+            if (raw_data.size() < min_size)
+                throw "Binary data is too small to be a 1D table.";
+
+            auto it_raw_data = raw_data.begin();
+
+            if (serialization::get_out<char>(it_raw_data) !=
+                static_cast<char>(sizeof(RealType))){
+                throw "Mismatch between RealType used to write and to read the 1D table";
+            }
+
+            m_x_min = serialization::get_out<RealType>(it_raw_data);
+            m_x_max = serialization::get_out<RealType>(it_raw_data);
+            m_x_size = m_x_max - m_x_min;
+            if(m_x_size < 0)
+                throw "raw_data contains invalid data.";
+
+            m_how_many_x = serialization::get_out<int>(it_raw_data);
+            if(m_how_many_x <= 0)
+                throw "raw_data contains invalid data.";
+            m_values = VectorType(m_how_many_x);
+            auto vals =
+                serialization::get_n_out<RealType>(it_raw_data, m_how_many_x);
+            std::copy(vals.begin(), vals.end(), m_values.begin());
+        };
+
+        /**
+        * Operator ==
+        *
+        * @param[in] rhs a const reference to a 1D table of the same type
+        * @return true if rhs and *this are equal. false otherwise
+        */
+        PXRMP_INTERNAL_GPU_DECORATOR PXRMP_INTERNAL_FORCE_INLINE_DECORATOR
+        bool operator== (
+            const equispaced_1d_table<RealType, VectorType> &rhs) const
+        {
+            return
+                (m_x_min == rhs.m_x_min) &&
+                (m_x_max == rhs.m_x_max) &&
+                (m_x_size == rhs.m_x_size) &&
+                (m_how_many_x == rhs.m_how_many_x) &&
+                (m_values == rhs.m_values);
+        }
+
+        /**
+        * Returns the i-th coodinate along x axis
+        *
+        * @param[in] i the index of the desired coordinate
+        * @return the i-th coordinate
+        */
+        PXRMP_INTERNAL_GPU_DECORATOR PXRMP_INTERNAL_FORCE_INLINE_DECORATOR
+        RealType get_x_coord(const int i) const noexcept
+        {
+                return i*m_x_size/(m_how_many_x-1) + m_x_min;
```

Comment:
Maybe add a member `m_dx` to avoid calculating `m_x_size/(m_how_many_x-1)` every time `get_x_coord` is called?

Reply:
Done!